plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -11,71 +11,89 @@
 
 class Test_WebP_Uploads_Picture_Element extends TestCase {
 
+	/**
+	 * Mime type.
+	 *
+	 * @var string
+	 */
+	public static $mime_type = 'image/webp';
+
 	/**
 	 * Attachment ID.
 	 *
 	 * @var int
 	 */
 	public static $image_id;
 
+	public function set_up(): void {
+		parent::set_up();
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => self::$mime_type ) ) ) {
+			$this->markTestSkipped( 'Mime type image/webp is not supported.' );
+		}
+
+		// Default to webp output for tests.
+		$this->set_image_output_type( 'webp' );
+
+		// Run critical hooks to satisfy webp_uploads_in_frontend_body() conditions.
+		$this->mock_frontend_body_hooks();
+	}
+
 	/**
 	 * Setup shared fixtures.
 	 */
 	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ): void {
+		// Fallback to JPEG IMG.
+		update_option( 'perflab_generate_webp_and_jpeg', '1' );
+
 		self::$image_id = $factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/data/images/leaves.jpg' );
 	}
 
+	public static function wpTearDownAfterClass(): void {
+		wp_delete_attachment( self::$image_id, true );
```

Comment:
The base class method should also always be called unless it's intended to completely override it.

```suggestion
		parent::wpTearDownAfterClass();
		wp_delete_attachment( self::$image_id, true );
```

The same should be done for `Test_Perflab_Server_Timing::wpTearDownAfterClass()`

Reply:
Hummm. Actually, this doesn't seem to be valid! I'm getting:

> Call to undefined method WP_UnitTestCase::wpTearDownAfterClass()

Reply:
Oh, I see. I was looking at `WP_Canonical_UnitTestCase` in core which [defines](https://github.com/WordPress/wordpress-develop/blob/4026237dadd2f4c1f11ba4a89605e8f394c11f38/tests/phpunit/includes/testcase-canonical.php#L22-L23) this method. With a quick glance I thought it was the base class but it isn't. The `tear_down_after_class()` static method in the base class has a [`method_exists()` check](https://github.com/WordPress/wordpress-develop/blob/4026237dadd2f4c1f11ba4a89605e8f394c11f38/tests/phpunit/includes/abstract-testcase.php#L84-L89) to see if this method exists.

So yeah, since the parent class lacks the method then no need for `parent::wpTearDownAfterClass();`!

Reply:
It will return error:

```
1) Test_WebP_Uploads_Picture_Element::tearDownAfterClass
Exception in Test_WebP_Uploads_Picture_Element::tearDownAfterClass
Call to undefined method WebP_Uploads\Tests\TestCase::wpTearDownAfterClass()

/var/www/html/wp-content/plugins/performance/plugins/webp-uploads/tests/test-picture-element.php:53
/wordpress-phpunit/includes/abstract-testcase.php:88
/var/www/html/wp-content/plugins/performance/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitGte8.php:107
```